TeamCode/src/main/java/t10/motion/hardware/PositionalMotor.java

```diff
@@ -4,52 +4,73 @@
+     * @param minBoundRotation the minimum inclusive rotation, in rotations, the motor will be allowed to rotate to (relative to initial position)
+     * @param maxBoundRotation the maximum inclusive rotation, in rotations, the motor will be allowed to rotate to (relative to initial position)
+     * @param initialRotation the initial position, in rotations, of the motor
+     * @param speed the speed for the motor to rotate at (can be negative for inverted rotation)
+     */
+    public PositionalMotor(DcMotorEx motor, float ticksPerRotation, float minBoundRotation, float maxBoundRotation, float initialRotation, float speed) {
+        if (minBoundRotation >= maxBoundRotation) {
+            throw new InvalidParameterException("minBoundRotation must be less than maxBoundRotation");
         }
 
         this.motor = motor;
-        this.ticksPerRevolution = ticksPerRevolution;
-        this.startPosition = startPosition;
-        this.currentPosition = startPosition;
-        this.endPosition = endPosition;
-        this.coefficient = coefficient;
+        this.ticksPerRotation = ticksPerRotation;
+        this.minBoundTicks = rotationToTicks(minBoundRotation);
+        this.maxBoundTicks = rotationToTicks(maxBoundRotation);
+        this.currentPositionTicks = rotationToTicks(initialRotation);
+        this.offsetTicks = rotationToTicks(initialRotation);
+
         this.speed = speed;
+
+        this.pid = new PIDController(0.1, 0, 0);
+    }
+
+    private float rotationToTicks(float rotation) {
+        return rotation * ticksPerRotation;
+    }
+
+    private float ticksToRotation(float ticks) {
+        return ticks / ticksPerRotation;
     }
 
-    public void setPosition(double position) {
-        if (position < startPosition || position > endPosition) {
-            throw new InvalidParameterException("Position is not within bounds [" + startPosition + "," + endPosition + "]");
+    public void setPosition(float rotation) {
+        if (rotation < ticksToRotation(minBoundTicks) || rotation > ticksToRotation(maxBoundTicks)) {
+            throw new InvalidParameterException("Position is not within bounds [" + ticksToRotation(minBoundTicks) + "," + ticksToRotation(maxBoundTicks) + "]");
         }
 
-        this.targetPosition = position;
-        this.startTicksPosition = motor.getCurrentPosition();
-        this.positionBeforeMovement = currentPosition;
+        this.targetTicks = rotationToTicks(rotation);
     }
 
     public void update() {
-        double revolutions = targetPosition - currentPosition;
-        double ticks = revolutions * ticksPerRevolution;
+        float ticks = targetTicks - currentPositionTicks;
 
-        motor.setVelocity(ticks * coefficient * speed);
+        motor.setVelocity(ticks * speed);
 
-        currentPosition = ((motor.getCurrentPosition() - startTicksPosition) / ticksPerRevolution) + positionBeforeMovement;
+        currentPositionTicks = motor.getCurrentPosition() + offsetTicks;
```

Comment:
this needs to be updated to use PID, for example: `double powerToSetMotorTo = this.pid.calculate(currentPosition, targetPosition)`

Reply:
**and this is POWER, NOT velocity